src/main.ts

```diff
@@ -360,17 +367,20 @@ export default class ReadwisePlugin extends Plugin {
             contentToSave = existingContent + contents;
           }
           await this.fs.write(originalName, contentToSave);
-          await this.removeBooksFromRefresh([bookID]);
         } catch (e) {
           console.log(`Readwise Official plugin: error writing ${processedFileName}:`, e);
           this.notice(`Readwise: error while writing ${processedFileName}: ${e}`, true, 4, true);
           if (bookID) {
             // handles case where user doesn't have `settings.refreshBooks` enabled
-            await this.addBookToRefresh(bookID);
+            await this.addToFailedBooks(bookID);

```

Comment:
ummm i think you need to exit early here right? otherwise the bookId will get removed from the list rigth after being added on line 381?

Reply:
(this wasn't the case until i suggested moving those remove calls outside the try/catch 😢

Reply:
f1d67e2

Reply:
also aligned this method with the other add/remove to refreshbooks fns WRT how it saves: 3092e8f